packages/dd-trace/src/dogstatsd.js

```diff
@@ -31,14 +43,26 @@ class Client {
     this._add(stat, value, 'c', tags)
   }
 
-  flush () {
+  flush (cb) {
     const queue = this._enqueue()
 
     if (this._queue.length === 0) return
 
     this._queue = []
 
-    if (this._family !== 0) {
+    if (this._httpOptions) {
+      requestQueue(queue, this._httpOptions, err => {
+        if (err) {
+          log.debug('HTTP error from agent: ' + err.stack)
+          this._httpOptions = null
+          this._queue = queue
+          this.flush()
```

Comment:
This is flushing only happening on error?

Reply:
The _recursive_ flush only happens on error. By setting the two properties above, it forces it to do UDP on the next attempt.